src/plugins/index_pattern_field_editor/public/components/field_editor/form_fields/script_field.tsx

```diff
@@ -61,13 +64,34 @@ export const ScriptField = ({ existingConcreteFields = [], links }: Props) => {
 
   const [{ type }] = useFormData<FieldFormInternal>({ watch: 'type' });
 
+  const validateScript = (setErrors: (errors: Array<ValidationError<string>>) => void) => {
```

Comment:
I wonder if this could live inside the `form_schema.ts` in a separate validation? (then you can decrease the `valueChangeDebounceTime: 0` on the field config which is set to 300ms by default if I am not wrong).

If I understand correctly there is only one `PainlessLang` singleton on the page to get the errors?

Reply:
That's a good point.

I think that would have been possible, but in my latest commit I made a change where you must provide the id of the editor you are fetching errors for. With this change, I don't think it will work inside of `form_schema.ts`. Let me know what you think.